swing/src/net/sf/openrocket/gui/components/DescriptionArea.java

```diff
@@ -202,5 +202,15 @@ public void setTextFont(Font font) {
 			editorPane.setFont(font);
 		}
 	}
+
+	public void setBackground(Color color) {
+		if (editorPane == null) return;
+		editorPane.setBackground(color);
+	}
+
+	public void setForeground(Color color) {
+		if (editorPane == null) return;
+		editorPane.setForeground(color);
```

Comment:
That doesn't really work on my machine (maybe it does on yours)?

Change this to:
```Java
public void setForeground(Color color) {
	editorPane.setForeground(color);
	StyledDocument styledDocument = (StyledDocument) editorPane.getDocument();
	SimpleAttributeSet attributeSet = new SimpleAttributeSet();
	StyleConstants.setForeground(attributeSet, color);
	styledDocument.setCharacterAttributes(0, styledDocument.getLength(), attributeSet, false);
}
```

and change `setText` to:
```Java
public void setText(String txt) {
	editorPane.setText(txt);
	editorPane.revalidate();
	SwingUtilities.invokeLater(new Runnable() {
		
		@Override
		public void run() {
			editorPane.scrollRectToVisible(new Rectangle(0, 0, 1, 1));
		}
		
	});
	setForeground(editorPane.getForeground());
	editorPane.scrollRectToVisible(new Rectangle(0, 0, 1, 1));
}
```

Also: the `if (editorPane == null) return;` condition is unnecessary. It's good that you try to be on the safe side, but in this scenario, it's just an unnecessary calculation.

Reply:
I kept the null condition check because on my machine, if I don't keep it my program crashes when I open a rocket model.